install_gridpack_deps.sh

```diff
@@ -1,133 +1,168 @@
-fi
+  popd || exit
+
+  echo "Boost installation complete"
+}
+
+function install_ga {
+  local ga_version=$1
+
+  # check args
+  : "${ga_version:?}"
 
-if ${install_ga}
-then
-  # Download, build, and install GA
-  cd ${GP_EXT_DEPS}
+  echo "--- Installing Global Arrays ${ga_version} ---"
 
-  echo "Downloading GA-5.8"
+  # download
+  echo "Downloading Global Arrays"
+  wget \
+    "https://github.com/GlobalArrays/ga/releases/download/v${ga_version}/ga-${ga_version}.tar.gz" \
+    -O ga.tar.gz \
+    --quiet
 
-  wget https://github.com/GlobalArrays/ga/releases/download/v5.8/ga-5.8.tar.gz
+  # unpack
+  echo "Unpacking Global Arrays"
+  tar -xf ga.tar.gz && rm ga.tar.gz
 
-  tar -xf ga-5.8.tar.gz
+  # remove version from dir name
+  mv "ga-${ga_version}" ga
 
-  cd ga-5.8
+  pushd ga || exit
 
-  # Build GA
-  echo "Building GA-5.8"
-  ./configure --with-mpi-ts --disable-f77 --without-blas --enable-cxx --enable-i4 --prefix=${PWD}/install_for_gridpack --enable-shared
-  
-  # Install GA
-  echo "Installing GA-5.8"
-  make -j 10 install
-  
-  echo "GA-5.8 installation complete"
-fi
+  # check if we are running on RHEL
+  distribution=$(
+    # shellcheck source=/dev/null
+    source /etc/os-release
+    echo "$ID"
+  )
+  case $distribution in
+  fedora | rhel | centos | rocky)
+    echo "Loading mpi4py module"
+    module load "mpi/openmpi-$(arch)"
+    ;;
+  esac
 
-if ${install_petsc}
-then
-    
-  # Install PETSc 3.16.4
-  cd ${GP_EXT_DEPS}
+  # build
+  echo "Configuring Global Arrays"
+  ./configure \
+    --with-mpi-ts \
+    --disable-f77 \
+    --without-blas \
+    --enable-cxx \
+    --enable-i4 \
+    --prefix="${PWD}/install_for_gridpack" \
+    --enable-shared
 
-  # Download
-  echo "Downloading PETSc 3.16.4"
+  # install
+  echo "Installing Global Arrays"
+  make -j "$(nproc)" install
 
+  popd || exit
+
+  echo "Global Arrays installation complete"
+}
+
+function install_petsc {
+  local petsc_version=$1
+
+  # check args
+  : "${petsc_version:?}"
+
+  echo "--- Installing PETSc ${petsc_version} ---"
+
+  # clone
+  echo "Cloning PETSc repository"
   git clone https://gitlab.com/petsc/petsc.git
-    
-  cd petsc
 
-  git checkout tags/v3.16.4 -b v3.16.4
+  pushd petsc || exit
+
+  git checkout "tags/v${petsc_version}" -b "v${petsc_version}"
 
   export PETSC_DIR=${PWD}
   export PETSC_ARCH=build-dir
 
-  # Install PETSc
-  echo "Installing PETSc 3.16.4"
-    
-  ./configure --download-superlu_dist --download-metis --download-parmetis --download-suitesparse --download-f2cblaslapack --download-cmake --prefix=${PWD}/install_for_gridpack --scalar-type=complex --with-shared-libraries=1 --download-f2cblaslapack=1
-
-  # Build PETSc
-  echo "Building PETSc 3.16.4"
+  # install
+  echo "Configuring PETSc"
+  ./configure \
+    --download-mumps \
+    --download-scalapack \
+    --download-metis \
+    --download-parmetis \
+    --download-suitesparse \
+    --download-f2cblaslapack \
+    --download-cmake \
```

Comment:
Why can't PETSc use the same system CMake as GridPACK does?  Consider testing without this option.

Reply:
This worked without `--download-cmake`